ape_hardhat/providers.py

```diff
@@ -431,6 +431,8 @@ class HardhatForkProvider(HardhatProvider):
     to use as your archive node.
     """
 
+    default_port = 8555
```

Comment:
I appreciate the intent here, but setting to a different default seems a little misleading. Could it detect not being a hardhat instance at 8545 and then change the port to start up? Like let's say you have hardhat connected to infura upstream

Reply:
Ah yes we can check the client version and change ports in that scenario, good idea

Reply:
This already happens:

```
➜  ERC4626 git:(chore/upgrade-ape) ✗ activate yoko
(yoko) ➜  ERC4626 git:(chore/upgrade-ape) ✗ ape console --network ethereum:local:hardhat
ERROR: (ProviderError) Port '8545' already in use by another process that isn't a Hardhat node.
```

So I am going to close this PR without merging.